spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java

```diff
@@ -30,14 +26,18 @@
  * @author Warren Zhu
  */
 @Configuration
-@EnableConfigurationProperties({AzureProperties.class, TelemetryProperties.class})
-@ConditionalOnClass(name = "com.microsoft.azure.management.Azure")
-@ConditionalOnProperty("spring.cloud.azure.credentialFilePath")
+@EnableConfigurationProperties(AzureProperties.class)
+@ConditionalOnClass(Azure.class)
+@ConditionalOnProperty("spring.cloud.azure.enabled")
```

Comment:
context should be always enabled ?

Reply:
Any auto config could be disabled. And user just provide their bean to continue use other dependent parts

Reply:
By default, we should enable all modules, right ?